storehaus-kafka/src/main/scala/com/twitter/storehaus/kafka/KafkaStore.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2014 Twitter inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.twitter.storehaus.kafka
+
+import java.util.Properties
+import java.util.concurrent.{Future => JFuture, TimeUnit}
+
+import com.twitter.storehaus.WritableStore
+import com.twitter.util.{Time, Future}
+import org.apache.kafka.clients.producer.{RecordMetadata, ProducerRecord, KafkaProducer}
+import org.apache.kafka.common.serialization.Serializer
+
+import scala.reflect.ClassTag
+
+/**
+  * Store capable of writing to a Kafka topic.
+  * @author Mansur Ashraf
+  * @since 11/22/13
+  */
+class KafkaStore[K, V](topic: String, props: Properties)
+  extends WritableStore[K, V] with Serializable {
+
+  private lazy val producer = new KafkaProducer[K, V](props)
+  private lazy val jFutureToTFutureConverter = {
+    val converter = new JavaFutureToTwitterFutureConverter
+    converter.start()
+    converter
+  }
+
+  /**
+    * Put a key/value pair in a Kafka topic
+    * @param kv (key, value)
+    * @return Future.unit
+    */
+  override def put(kv: (K, V)): Future[Unit] =
+    putAndRetrieveMetadata(kv).map(_ => ())
+  
+  /**
+    * Put a key/value pair in a Kafka topic and retrieve record metadata
+    * @param kv (key, value)
+    * @return Future[RecordMetadata]
+    */
+  def putAndRetrieveMetadata(kv: (K, V)): Future[RecordMetadata] = jFutureToTFutureConverter {
+    val (key, value) = kv
+    producer.send(new ProducerRecord[K, V](topic, key, value))
+  }
+
+  /**
+    * Close this store and release any resources.
+    * It is undefined what happens on put/multiGet after close
+    */
+  override def close(time: Time): Future[Unit] = Future {
+    producer.close()
```

Comment:
we should probably stop the thread converting java to twitter futures (and complete all the open promises with some exception).